ouroboros-consensus-cardano-test/test/Test/ThreadNet/Cardano.hs

```diff
@@ -630,26 +629,28 @@ prop_simple_cardano_convergence TestSetup
         overlayOffsets i =
             -- Shift to account for the initial Byron era.
             Set.mapMonotonic (Util.addSlots numByronSlots) $
-
-            Map.keysSet $
-            SL.overlayScheduleToMap $
-            runIdentity $ flip runReaderT shelleyGlobals $
-            SL.overlaySchedule
-              @(ShelleyEra Crypto)
-              (EpochNo i)   -- NB 0 <-> the first /Shelley/ epoch
-              genesisKeyHashes
-              (sgProtocolParams genesisShelley)   -- notably contains setupD
-
-        shelleyGlobals =
-            SL.mkShelleyGlobals
-              genesisShelley
+            -- Note: this is only correct if each epoch uses the same value for @d@
+            overlaySlotsInEpoch
               -- Suitable only for this narrow context
               (fixedSizeEpochInfo epochSizeShelley)
-              (getMaxMajorProtVer maxMajorPVShelley)
-
-        genesisKeyHashes =
-            Set.fromList $
-            map (Shelley.mkKeyHash . Shelley.cnGenesisKey) coreNodes
+              -- notably contains setupD
+              (SL._d (sgProtocolParams genesisShelley))
+                -- NB 0 <-> the first /Shelley/ epoch
+              (EpochNo i)
+
+        overlaySlotsInEpoch ::
+             EpochInfo Identity
+          -> SL.UnitInterval
+          -> EpochNo
+          -> Set SlotNo
+        overlaySlotsInEpoch ei d epoch = runIdentity $ do
+            epochSize <- epochInfoSize  ei epoch
+            firstSlot <- epochInfoFirst ei epoch
+            let slotsInEpoch =
+                   [firstSlot .. Util.addSlots (unEpochSize epochSize) firstSlot]
+            return
+              $ Set.fromList
+              $ filter (SL.isOverlaySlot firstSlot d) slotsInEpoch
```

Comment:
I have asked that ledger-specs provides a function like this (not a `Set`, but a list). This is just a very naive (linear!) implementation.

@nfrisby This is just used as a label, not a check. So what if I got it wrong?

Reply:
> I have asked that ledger-specs provides a function like this (not a `Set`, but a list). This is just a very naive (linear!) implementation.

That's what I was expecting to add to ledger-specs, is it too naive? ie just filtering the epoch slots with `isOverlaySlot`.

Alternatively, you can invert `isOverlaySlot` and compute them directly. which would be `floor (x / d)`. I thought since it was testing it wouldn't matter too much, and having `isOverlaySlot` as the single source of truth would be nice.

Reply:
Since this is just testing, either way is fine. As long as `overlaySlotsInEpoch` (feel free to pick another name) is defined in ledger-specs and not here. If it's a naive linear filter, then there should be a warning in the docstring so that nobody would ever consider using it for non-testing code.

Reply:
> So what if I got it wrong?

To validate, I'd run the test with `--iohk-enable-nightly-tests` and with 200 iterations (~30hr on a CI box) or more(depending on how much time you have. As long as you see `True` and `False` for `label_hadActiveNonOverlaySlots`, then it's behaving as expected. (This test is ultimately statistical, so I left it as a manual and occasional check for now, as we work out our statistical testing plan.)